apps/rebar/src/rebar_prv_manifest.erl

```diff
@@ -0,0 +1,138 @@
+%% ===================================================================
+%% Manifest Provider
+%% ===================================================================
+-module(rebar_prv_manifest).
+-behaviour(provider).
+-export([init/1,
+         do/1,
+         format_error/1]).
+
+-include_lib("providers/include/providers.hrl").
+
+-define(PROVIDER, manifest).
+-define(DEFAULT_FORMAT, erlang).
+
+-type extension() :: string().
+-type app_context() :: #{name         := binary(),
+                         src_dirs     := [file:filename()],
+                         include_dirs := [file:filename()],
+                         src_ext      := extension(),
+                         out_mappings := [#{extension := extension(), path := file:filename()}],
+                         dependencies_opts => any()}.
+-type manifest() :: #{
+                      apps := [app_context()],
+                      deps := [app_context()],
+                      otp_lib_dir := string(),
+                      source_root := string()
+                     }.
+
+-type format() :: erlang | eetf.
+
+%% ===================================================================
+%% Provider Callbacks
+%% ===================================================================
+-spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
+init(State) ->
+
+  %% By default, the provider outputs the manifest to stdout, so disable logs
+  %% not to interfere.
+  ok = rebar_log:init(api, 0),
+
+  State1 = rebar_state:add_provider(
+             State,
```

Comment:
for formatting rules, Rebar3 has historically used 4 spaces indents everywhere in providers. As per the nesting, it tends to look more like https://github.com/erlang/rebar3/blob/main/apps/rebar/src/rebar_prv_repos.erl -- this is not a blocker, but generally preferable.